File: 0061.RotateList.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution(object):
    def rotateRight(self, head, k):
        """
        :type head: ListNode
        :type k: int
        :rtype: ListNode
        """
        if not head:
            return head
        l = 1
        tail = head
        while tail.next:
            l += 1
            tail = tail.next
        tail.next = head
        k %= l
        for i in xrange(l-k):
            tail = tail.next
        head = tail.next
        tail.next = None
        return head
        

File: 0000-0999/0061.RotateList.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
 func rotateRight(head *ListNode, k int) *ListNode {
    if head == nil {
        return head
    }
    tail, n := head, 1
    for tail.Next != nil {
        tail = tail.Next
        n++
    }
    k %= n
    if k == 0 {
        return head
    }
    tail.Next = head
    for i := 0; i < n-k; i++ {
        tail = tail.Next
    }
    newHead := tail.Next
    tail.Next = nil
    return newHead
}